### DISCUSSION TITLE
Ruby on Rails support?

### DISCUSSION THREAD

**Discussion by reynardmh** [ddev/ddev#6338] [General]
Is there any plan to implement Rails support?  I'm interested in contributing as a side project while I also learn go. Any advice / guidance would be very much appreciated.

**Comment by tyler36** [comment]
Hi @reynardmh 

I'm not aware of any talk to add Ruby on Rails support but PRs are always welcome.

I would recommend trying to get DDEV working with Rails on the "php" project type first. What settings do you need to change/add? Document as you go.
The next step would then be adding the project type to DDEV. 
- see [Adding new project types](https://ddev.readthedocs.io/en/latest/developers/project-types/)
- check out contributing training session for guides on setup etc. https://ddev.com/blog/contributor-training/

You might want to read over the PRs for "recently" added project types:
- [feat: add Silverstripe project type](https://github.com/ddev/ddev/pull/5119)
- [Adds Craft CMS project type, docs, and tests closes #3594 #4176](https://github.com/ddev/ddev/pull/4176)

If you need help, just ask on our DDEV discord channel (here's an [invite](https://discord.gg/5wjP76mBJD))

**Reply by rfay** [reply]
@tyler36 is right that there are no plans to explicitly support Rails. He's also right that it could be just running the rails app inside DDEV, just like you would do with node.js. https://ddev.readthedocs.io/en/stable/users/extend/customization-extendibility/#running-extra-daemons-in-the-web-container

**Reply by reynardmh** [reply]
Thank you all, I'll start with the resources you shared and dive into the code base to understand it more.

**Reply by rfay** [reply]
The first level of approach has nothing to do with exploring code. You'll just use web_extra_daemons and web_extra_exposed_ports to run rails, and use webimage_extra_packages to add whatever ruby packages you need.

**Reply by reynardmh** [reply]
Thanks for the tips, I'll try that. I can't promise fast turnaround. Just looking for a side project while I learn go. So understanding the code base is actually high on my list :)

**Reply by rfay** [reply]
We welcome you!

**Reply by rfay** [reply]
You've seen the various tutorials in https://ddev.com/blog/contributor-training/, right? And of course the docs, https://ddev.readthedocs.io/en/stable/developers/building-contributing/

**Reply by reynardmh** [reply]
Not yet. That's great you have contrib training. I'll go through those resources. Thanks!

**Reply by tyler36** [reply]
Once you know the manual steps required, then you know what needs to change in DDEV.
